File: learn-api/app/schemas/questionnaire_answer_statistic.py
from typing import Optional

from pydantic import BaseModel

from app.schemas.questionnaire_question_option import QuestionnaireQuestionOption
from app.schemas.user import User


class QuestionnaireAnswerStatistic(BaseModel):
    id: Optional[int]
    answer: Optional[str]
    selected: Optional[str]
    question_option: Optional[QuestionnaireQuestionOption]
    user: Optional[User]
    questionnaire_id: Optional[int]

File: frontend/src/model/questionnaires/questionnaireAnswerStatistic.ts
import { User } from '../user';
import { QuestionnaireAnswer } from './questionnaireAnswer';
import { QuestionnaireQuestionOption } from './questionnaireQuestionOption';

export interface QuestionnaireAnswerStatistic {
  id: number;
  answer: string;
  selected: string;
  question_option: QuestionnaireQuestionOption;
  user: User;
  questionnaire_id: number;
}
